craft_archives/repo/apt_key_manager.py

```diff
@@ -149,6 +157,8 @@ def install_key(self, *, key: str) -> None:
         except subprocess.CalledProcessError as error:
             raise errors.AptGPGKeyInstallError(error.output.decode(), key=key)
 
+        # Change the permissions on the file so that APT itself can read it later
+        os.chmod(self._gpg_keyring, 0o644)
```

Comment:
Why not?

```suggestion
        self._gpg_keyring.chmod(0o644)
```

Reply:
no reason; the NamedTemporaryFile is a string so I just used os.chmod() for all of them
now updated all of the to Path.chmod(), thanks